### PR TITLE
Upgrade anonymizer nuget

### PR DESCRIPTION
**Pull request by tongwu-msft**
Upgrade anonymizer nuget to use FHIR lib 3.4

## Testing
Enable anonymized export tests in CI pipeline.

## FHIR Team Checklist
- [x] **Update the title** of the PR to be succinct and less than 50 characters
- [x] **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- [x] Tag the PR with the type of update: **Bug**, **Dependencies**, **Enhancement**, or **New-Feature**
- [x] Tag the PR with **Azure API for FHIR** if this will release to the managed service
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Patch|Skip|Feature|Breaking (reason)

### CODE REVIEW

File: build/jobs/run-pr-tests.yml [hunk 1]
```diff
@@ -98,6 +98,15 @@ jobs:
         Write-Host "##vso[task.setvariable variable=TestContainerRegistryServer]$($acrLoginServer)"
         Write-Host "##vso[task.setvariable variable=TestContainerRegistryPassword]$($acrPassword)"
 
+        $exportStoreSettings = $appSettings | where {$_.Name -eq "FhirServer__Operations__Export__StorageAccountUri"}
```
Comment (Ivanidzo4ka) [inline build/jobs/run-pr-tests.yml:101]: [https://github.com/microsoft/fhir-server/blob/de304363c71749dc993f6b5fa9076b55b6561e32/samples/templates/default-azuredeploy.json#L235](https://github.com/microsoft/fhir-server/blob/de304363c71749dc993f6b5fa9076b55b6561e32/samples/templates/default-azuredeploy.json#L235)
or 
[https://github.com/microsoft/fhir-server/blob/de304363c71749dc993f6b5fa9076b55b6561e32/samples/templates/default-azuredeploy-docker.json#L231](https://github.com/microsoft/fhir-server/blob/de304363c71749dc993f6b5fa9076b55b6561e32/samples/templates/default-azuredeploy-docker.json#L231)

need to be updated to include 
FhirServer__Features__SupportsAnonymizedExport

Comment (tongwu-msft) [reply build/jobs/run-pr-tests.yml:101]: yes, this option previous major to disable for PaaS. Update the default value to true.

**Review by Ivanidzo4ka — APPROVED** [review]
:shipit:

**Comment by brendankowitz** [comment]
Thanks @tongwu-msft, @Ivanidzo4ka!

**Pull request merged**